src/understand-tidb/execution.md

```diff
@@ -1 +1,27 @@
+
+## Execution Framework
+
+TiDB builds the computing engine based on the distributed storage provided by TiKV. The TiKV server implements a coprocessor framework to support distributed computing. The computing operations will be pushed to the TiKV coprocessor as far as possible to accelerate the computation speed. That is to say, a sub-plan of the SQL execution plan will be parallelly executed on different TiKV servers, and the result of each sub-plan will be collected to a TiDB server to compute for the final result. 
+
+The processing model of the execution plan tree is known as the Volcano iterator model. The essential of the Volcano model is abstracted to 3 interfaces: `Open`, `Next`, `Close`. All operators offer the same interfaces and the implementation is opaque. `Open` will be invoked in turn for each operator to init the needed resources before computing. Conversely, `Close` will release the resources. To obtain the query output, the final operator in the plan tree will keep invoking `next` until no tuple is pulled from its child.
+
+It's easy to understand how the Volcano model works for single-process execution. For parallism issues, the Volcano introduces an operator called `exchange` at any desired point in a plan tree. Further explanation about the parallism-related issues would be introduced in the [execution-framework.md](execution-framework.md) section.
+
+## Vectorized Execution
+
+Like the Volcano iteration model, vectorization uses pull-based (root-to-leaf traversal) iteration where each operator has a `next` method that produces result tuples. However, each `next` call fetches a block of tuples instead of just one tuple. The main principle of vectorized execution is batched execution on a columnar data representation:  every “work” primitive function that manipulates data does not work on a single data item, but on a vector (an array) of such data items that represents multiple tuples. The idea behind vectorized execution is to amortize the iterator call overhead by performing as much as possible inside the data manipulation methods. For example, this work can be to hash 1000s of values, compare 1000s of string pairs, update a 1000 aggregates, or fetch a 1000 values from 1000s of addresses.
+
+Columnar Different from the row-oriented data representation, columnar format organize data by column rather by row. By storing data in columns rather than rows, the database can more precisely access the data it needs to answer a query rather than scanning and discarding unwanted data in rows. The memory columnar data representation in TiDB is defined as `Chunk`, which is inspired by [Apache Arrow](https://arrow.apache.org/).  The detailed definition and usage of `Chunk` will be introduced in the [implementation-of-vectorized-execution.md](implementation-of-vectorized-execution.md) section.
```

Comment:
```suggestion
Columnar Different from the row-oriented data representation, columnar format organize data by column rather by row. By storing data in columns rather than rows, the database can more precisely access the data it needs to answer a query rather than scanning and discarding unwanted data in rows. The memory columnar data representation in TiDB is defined as `Chunk`, which is inspired by [Apache Arrow](https://arrow.apache.org/).

The detailed definition and usage of `Chunk` will be introduced in the [Implementation of Vectorized Execution](implementation-of-vectorized-execution.md) section.
```